### PR TITLE
added new color palette schemes

### PR DESCRIPTION
**Pull request by nbokulich**
Most of the default seaborn color schemes really don't work well for the classification confusion matrix heatmaps. The reason is that these heatmaps contain many values at the extremes and fewer intermediates values, and some colormaps have very bold colors, very dark colors, or black/white at both termini. With the wrong color palette, the results can make your eyes bleed.

So I made my own custom color palettes with softer termini and cheesier names.

Also added a unit test to verify that all color palette names are valid input for `sns.heatmap`.

q2-sample-classifier-color-schemes

### CODE REVIEW

File: q2_sample_classifier/tests/test_classifier.py [hunk 1]
```diff
@@ -240,6 +241,13 @@ def test_coordinates_format_to_metadata(self):
                         name='Latitude', index=exp_index)
         self.assertEqual(sorted(exp), sorted(obs_category.to_series()))
 
+    # this just checks that palette names are valid input
+    def test_custom_palettes(self):
+        confused = np.array([[1, 0], [0, 1]])
+        for cm in ['Default', *_custom_palettes().keys()]:
```
Comment (ebolyen) [inline q2_sample_classifier/tests/test_classifier.py]: You use `cm` for a confusion matrix elsewhere, could this be renamed to `pallete`?

File: q2_sample_classifier/tests/test_classifier.py [hunk 2]
```diff
@@ -240,6 +241,13 @@ def test_coordinates_format_to_metadata(self):
...
+            print(cm)
```
Comment (ebolyen) [inline q2_sample_classifier/tests/test_classifier.py]: remove print

File: q2_sample_classifier/plugin_setup.py [hunk 3]
```diff
@@ -217,12 +218,7 @@ def _6(ff: BooleanSeriesFormat) -> (qiime2.Metadata):
              'GradientBoostingClassifier', 'AdaBoostClassifier',
              'KNeighborsClassifier', 'LinearSVC', 'SVC']),
         'palette': Str % Choices(
-            ['Default', 'Greys', 'Purples', 'Blues', 'Greens', 'Oranges',
-             'Reds', 'YlOrBr', 'YlOrRd', 'OrRd', 'PuRd', 'RdPu', 'BuPu',
-             'GnBu', 'PuBu', 'YlGn', 'bone_r', 'pink_r', 'summer_r', 'cool',
-             'hot_r', 'copper_r', 'gist_earth_r', 'terrain_r', 'gnuplot2_r',
-             'cubehelix_r', 'jet_r', 'viridis_r', 'plasma_r', 'inferno_r',
-             'magma_r'])},
+            ['Default', *_custom_palettes().keys()])},
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_sample_classifier/plugin_setup.py]: 👍

File: q2_sample_classifier/visuals.py [hunk 4]
```diff
@@ -145,17 +194,24 @@ def _linear_regress(actual, pred):
                         index=[actual.name])
 
 
+def _plot_heatmap_from_confusion_matrix(cm, palette):
+    if palette == 'Default':
+        palette = _custom_palettes()['sirocco']
+    else:
+        palette = _custom_palettes()[palette]
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_sample_classifier/visuals.py]: you have a point (the function is now 2 lines) but I'd prefer to keep it as a separate function to contain any other modifications that are made to the heatmap plots in the future

**Comment by nbokulich** [comment]
thanks @ebolyen ! made all changes as requested.

**Pull request merged**